tests/integration_tests/scheduler/test_generic_driver.py

```diff
@@ -106,3 +108,39 @@ async def finished(iens, returncode):
     await driver.submit(0, "sh", "-c", "sleep 60; exit 2", name=job_name)
     await poll(driver, {0}, started=started, finished=finished)
     assert aborted_called
+
+
+async def test_kill_actually_kills(driver: Driver, tmp_path, pytestconfig):
+    os.chdir(tmp_path)
+
+    if (isinstance(driver, LsfDriver) and pytestconfig.getoption("lsf")) or (
+        isinstance(driver, OpenPBSDriver) and pytestconfig.getoption("openpbs")
+    ):
+        # Allow more time when tested on a real compute cluster to avoid false positives.
+        job_kill_window = 10
+        test_grace_time = 20
+    elif sys.platform.startswith("darwin"):
+        # Mitigate flakiness on low-power test nodes
+        job_kill_window = 5
+        test_grace_time = 8
+    else:
+        job_kill_window = 1
+        test_grace_time = 2
+
+    async def started(iens):
+        nonlocal driver
+        await driver.kill(iens)
+
+    await driver.submit(
+        0,
+        "sh",
+        "-c",
+        f"sleep {job_kill_window}; touch {tmp_path}/survived",
+        name="kill_me",
+    )
+    await poll(driver, {0}, started=started)
+
+    # Give the script a chance to finish if it is running
+    await asyncio.sleep(test_grace_time)
+
+    assert not Path("survived").exists(), "Job should have been killed"
```

Comment:
How does this guarantee that the job actually have run; ie. submitted correctly? Maybe like to touch a file `file_start` first (before sleep) and then check that the `file_start` is there while `survived` is not?

Reply:
This is ensured by `await poll(driver, {0}, started=started)`